### ISSUE TITLE
Align JDK 16 compiler toolchains with upstream

### ISSUE THREAD

**Issue by aahlenst** [AdoptOpenJDK/openjdk-build#2432]
Sources:

* https://wiki.openjdk.java.net/display/Build/Supported+Build+Platforms
* `strings -a bin/javac | grep "GCC:"` on upstream binaries

Actions:
* [ ] Upgrade JDK16+ to VS 2019 on Windows x86_64 and x86_32
* [ ] Upgrade JDK 16+ to gcc 10.x on Linux x86_64 (10.2 is currently in use; Alpine is still on 8.3)
* [ ] Upgrade JDK 16+ to gcc 10.x on Linux aarch64 (10.2 is currently in use)

Unknown:
* [ ] Check whether JDK16+ use XLC/xlclang++ 16.1 on AIX ppc; otherwise upgrade.

Already okay:
* JDK16+ uses gcc 7.5 on Linux arm32, ppc64le, s390x. (However, a brave soul might trying a newer version)

**Comment by pshipton** [comment]
Does this affect OpenJ9 builds? From an OpenJ9 perspective:
- Windows VS2019 yes - https://github.com/AdoptOpenJDK/openjdk-build/issues/2147
- OpenJ9 switching to gcc 10.x hasn't been tried, not sure if it can build, or if it will introduce any problems, or how it will affect performance.
- JDK16+ already uses xlclang 16 on AIX

**Comment by aahlenst** [comment]
> Does this affect OpenJ9 builds?

No, that was not the plan. But thanks for the info nonetheless.

**Comment by aahlenst** [comment]
We cannot simply upgrade to GCC 10.2 because it cannot be compiled on RHEL/CentOS 6 anymore:

```
./configure --prefix=/usr/local/gcc10
...
configure: error: Building GCC requires GMP 4.2+, MPFR 3.1.0+ and MPC 0.8.0+.
```

The versions of GMP, MPFR, and MPC bundled with RHEL 6/CentOS 6 are too old. RHEL/CentOS 7 is fine.

**Comment by sxa** [comment]
Took a while, but we're in the process of setting up GCC 10.3 under https://github.com/adoptium/temurin-build/issues/2787 - this seems to produce a usable build of OpenJDK for versions 8 and 17. Adjustments to the build to utilise it are expected to wait until after the January releases.

**Comment by sxa** [comment]
Final step on this will be https://github.com/adoptium/temurin-build/pull/2861
